src/main/java/io/vertx/ext/mail/SMTPException.java

```diff
@@ -28,20 +28,128 @@
  */
 public class SMTPException extends NoStackTraceThrowable {
 
+  /**
+   * The Enhanced Status codes.
+   *<p>
+   * See: https://datatracker.ietf.org/doc/html/rfc3463#section-3
+   *</p>
+   */
+  public enum EnhancedStatus {
+
+    /**
+     * <b>x.0.x</b> for Undefined errors
+     */
+    OTHER_UNDEFINED(0),
+    /**
+     * <b>x.1.x</b> for Address Status errors
+     */
+    OTHER_ADDRESS(1),
+    /**
+     * <b>x.2.x</b> for Mailbox Status errors
+     */
+    OTHER_MAILBOX(2),
+    /**
+     * <b>x.3.x</b> for Mail System errors
+     */
+    OTHER_MAIL_SYSTEM(3),
+    /**
+     * <b>x.4.x</b> for Network and Routing errors
+     */
+    OTHER_NETWORK(4),
+    /**
+     * <b>x.5.x</b> for Mail Delivery Protocol errors
+     */
+    OTHER_MAIL_DELIVERY(5),
+    /**
+     * <b>x.6.x</b> for Message Content or Message Media errors
+     */
+    OTHER_MAIL_MESSAGE(6),
+    /**
+     * <b>x.7.x</b> for Security or Policy errors
+     */
+    OTHER_SECURITY(7),
+    /**
+     * Anything else is unknown, like failures happen before knowing the capabilities or smtp server reply maliciously.
+     */
+    OTHER_UNKNOWN(-1);
+
+    private final int subject;
+    private int detail;
```

Comment:
this should be final

Reply:
No, because `subject` is enumerable, but `detail` is not.